src/network/networkd-sysctl.c

```diff
@@ -5,15 +5,190 @@
 #include <linux/if_arp.h>
 
 #include "af-list.h"
+#include "fd-util.h"
 #include "missing_network.h"
 #include "networkd-link.h"
 #include "networkd-manager.h"
 #include "networkd-network.h"
 #include "networkd-sysctl.h"
+#include "path-util.h"
 #include "socket-util.h"
 #include "string-table.h"
 #include "sysctl-util.h"
 
+#if HAVE_VMLINUX_H
+
+#include "bpf/sysctl_monitor/sysctl-monitor-skel.h"
+#include "bpf/sysctl_monitor/sysctl-write-event.h"
+
+#define CGROUP_MOUNT_DFLT "/sys/fs/cgroup"
+
+static int open_rootcg(void) {
+        int fd = open(CGROUP_MOUNT_DFLT, O_PATH | O_DIRECTORY | O_CLOEXEC);
```

Comment:
hmm, please use cg_get_root_path()

Reply:
It doesn't use O_PATH.
It's not a big deal, but I would prefer using it somehow.

Reply:
Ah I see, just to get the CGROUP_MOUNT_DFLT value.
Will do

Reply:
I don't understand what's going on, `cg_get_root_path()` returns an empty string for me.
This simple snippet:
```c
_cleanup_free_ char *rootcg = NULL;
int fd, r;

r = cg_get_root_path(&rootcg);
if (r < 0)
        return log_error_errno(r, "Failed to get root cgroup path: %m");

log_error("rootcg: '%s', r: %d", rootcg, r);
```
Produces this output:
```
rootcg: '', r: 0
```

Reply:
I solved it by using:
```c
cg_path_open(SYSTEMD_CGROUP_CONTROLLER, "/")
```
Even if I don't really like that the fd is opened without `O_PATH`

Reply:
it generates a path that isn't suffixed with a slash basically. (which means for the root it returns "")-

you can use empty_to_root() around cg_get_root_path()'s returned path to turn the empty string to "/".

Reply:
> ```c
> cg_path_open(SYSTEMD_CGROUP_CONTROLLER, "/")
> ```

The patch doesn't show that?

Reply:
I didn't push yet, sorry.

Reply:
Since https://github.com/kernel-patches/bpf-rc/commit/7f6287417baf57754f47687c6ea1a749a0686ab0 was merged, I can rework the bpf program to filter the events in the kernel